Updating Collision errors article (#1679)
umbraco/UmbracoDocs: Umbraco-Cloud/Troubleshooting/Deployments/Structure-Error/index.md

In this example there are two `.uda` files who share the same alias which leads to a conflict: it is impossible for Deploy to know which of the files to use, so it gives up and sends an error back.

:::note
Does the collision error involve **Dictionary items**?
Use this guide instead: [Troubleshooting duplicate dictionary items](../Duplicate-Dictionary-Items)
:::

You can run into an error like this on all of your Cloud environment. Somestimes you might also run into it, on a local clone of your project. This guide will use an example, where two files are colliding on a Development and a Live environment.

## Table of content

* [Cause](#cause)
* [Video tutorial](#video-tutorial)
* [Using the error message](#using-the-error-message)
* [Deciding which file you want to use](#deciding-which-file-you-want-to-use)
* [Getting your environments in sync](#getting-your-environments-in-sync)

## Cause

<iframe width="800" height="450" src="https://www.youtube.com/embed/pF5SUh30FKI?rel=0" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>

The main cause of this problem is when an entity has been manually created in two or more environments, using the same alias. 

> Each environment on Umbraco Cloud are isolated and do not know what the other one is doing until they are syncronised. When an entity is created with identical aliases on two or more environments, it will create duplicate entities that are considered separate entities even though they look that same and share the same alias on both/all environments.

Reviewer: ```suggestion
Each environment on Umbraco Cloud are isolated and do not know what the other one is doing until they are synchronized. When an entity is created with identical aliases on two or more environments, it will create duplicate entities that are considered separate entities even though they look that same and share the same alias on both/all environments.
```

Let's use the example from the beginning of this article, where two `.uda` files for the Document Type "home" are colliding.

    Some artifacts collide on unique identifiers.
    This means that they have different Udis, yet
    they refer to the same unique Umbraco object
    and therefore cannot be processed.
    ---------------------------------------------
    Collisions for entity type "document-type": 
      Collisions for unique identifier "home":
        UdaFile: ~/data/revision/document-type__4c04d968448747d791b5eae254afc7ec.uda
        UdaFile: ~/data/revision/document-type__f848c577f02b4ee5aea84f87458072a4.uda

For this example, it’s decided that the Document Type currently used on the Live environment is the one we want to use going forward.

![Visual representation of the error message](images/visualizing.png) - *A visual representation of the error message*

In order to figure out which of the two colliding `.uda` files is the one for the Document Type being used on the Live environment follow these steps:

1. Access **Kudu** on the Live environment
2. Use the CMD console (found under the 'Debug console' menu) to navigate to your `site/wwwroot/data/revision` folder

> 3. Remove the colliding `.uda` files mentioned in the error message

Reviewer: ```suggestion
3. Remove both of the colliding `.uda` files mentioned in the error message
```

Author: I've seen cases where there are more than 2 🤷‍♀️

4. Go back to the `/wwwroot/data` folder and run this command: `echo > deploy-export` in the console

> 5. This will regenerate a `.uda` file for the Document Type being used on the Live environment

Reviewer: ```suggestion
5. This will regenerate all `.uda` files for the Live environment - this means only the currently used one will be there afterwards
```

6. Run the command: `echo > deploy` in the same folder, to make sure everything is extracting correctly

![Finding correct UDA file](images/visualizing-2.png)

You now know which `.uda` file you want. In this case: `document-type__1.uda`.

## Removing the unused file

> We strongly recommend that you resolve this locally since this will ensure that the changes you make are added to your Git repositories.

Reviewer: ```suggestion
:::warning
We strongly recommend that you resolve this locally since this will ensure that the changes you make are added to your Git repositories. Otherwise you may end up having the same problem next time you deploy!
:::
```
